### PR TITLE
[acl] Use a list instead of a comma-separated string for ACL port list

### PR DESCRIPTION
**Pull request by daall**
Signed-off-by: Danny Allen <daall@microsoft.com>

#### What I did
Fixes https://github.com/Azure/sonic-buildimage/issues/4934

#### How I did it
I discovered that using the @ syntax to insert the port list as a comma-separated list/array actually causes the port list to be deleted when the same table is `config add`-ed multiple times. Using a list works. (We still need to determine the root cause for why the @ syntax fails).

#### How to verify it
Updated the tests and ran the command locally to verify the fix. Still working on adding a new regression test case for this.

### CODE REVIEW

File: config/main.py [hunk 1]
```diff
@@ -3349,7 +3349,7 @@ def parse_acl_table_info(table_name, table_type, description, ports, stage):
         if port not in valid_acl_ports:
             raise ValueError("Cannot bind ACL to specified port {}".format(port))
 
-    table_info["ports@"] = ",".join(port_list)
+    table_info["ports"] = list(port_list)
```
Comment (qiluo-msft) [inline config/main.py:3352]: >list [](start = 26, length = 4)

It's already a list, right?

File: tests/acl_config_test.py [hunk 2]
```diff
@@ -36,7 +34,7 @@ def test_parse_table_with_vlan_and_duplicates(self):
         assert table_info["stage"] == "egress"
 
         # Since Ethernet4 is a member of Vlan1000 we should not include it twice in the output
-        port_list = table_info["ports@"].split(",")
+        port_list = set(table_info["ports"])
```
Comment (qiluo-msft) [inline tests/acl_config_test.py:37]: >port_list [](start = 8, length = 9)

port_set?

**Comment by lguohan** [comment]
unit itest?

**Comment by daall** [comment]
> unit itest?

I updated the existing unit tests to reflect this change. I don't think it's possible to unit test this exact scenario b/c the mock config db APIs will overwrite the swsscommon operation that is causing the issue.

**Comment by qiluo-msft** [comment]
Could you add a vs test instead? If that is challenging, a kvm test is the next choice, and finally sonic-mgmt test case.

**Comment by daall** [comment]
> Could you add a vs test instead? If that is challenging, a kvm test is the next choice, and finally sonic-mgmt test case.

I can add a vs test!

File: tests/acl_config_test.py (hunk 2 above)
Comment (qiluo-msft) [reply tests/acl_config_test.py:37]: I mean the variable name is not proper. Not a blocking issue, but you can make it better readable.

---
In reply to: [598998736](https://github.com/Azure/sonic-utilities/pull/1519#discussion_r598998736) [](ancestors = 598998736)

**Review by qiluo-msft — APPROVED** [review]

**Comment by daall** [comment]
> > Could you add a vs test instead? If that is challenging, a kvm test is the next choice, and finally sonic-mgmt test case.
> 
> I can add a vs test!

https://github.com/Azure/sonic-swss/pull/1694

**Comment by qiluo-msft** [comment]
@daall, I tried cherry-pick to 201911 but see conflicts in unit test code. Could you submit another PR on 201911 branch?

**Pull request merged**